src/v/raft/tests/type_serialization_tests.cc

```diff
@@ -220,7 +220,7 @@ SEASTAR_THREAD_TEST_CASE(heartbeat_response_roundtrip) {
             model::node_id(2), model::revision_id(i)),
           .node_id = raft::vnode(model::node_id(1), model::revision_id(i)),
           .group = raft::group_id(i),
-          .term = model::term_id(random_generators::get_int(-1, 1000)),
+          .term = model::term_id(random_generators::get_int(0, 1000)),
```

Comment:
if this is fixing something with decoding, then it seems like the problem is in the decoder not here since `-1` as a meaning related to optional is only relevant when decoding optionals. I'm curious what is the relationship between this test and jitter_test.cc?

Reply:
the decoding is really complicated. it almost looks to me like the model::offset is being encoded as a normal fixed-size integer, but decoded as a variable-size encoded integer.

Reply:
indeed the test was wrong, both offset and meta may be negative, they are replaced by `-1`  in encoded version then when decoded they are replacing with min numeric limit for a given value.

Reply:
I would add an explicit test case for `model::offset::min()`